tests/test_deployments/test_app.py

```diff
@@ -269,3 +275,54 @@ def get_missing(messages: List[str]) -> List[str]:
         assert not get_missing(before_stopping), get_logs()
         self.tendermint_node.stop()
         assert not get_missing(after_stopping), get_logs()
+
+
+class TestTendermintBufferFailing(BaseTendermintServerTest):
+    """Test Tendermint buffer"""
+
+    perform_monitoring = False  # Setting this flag to False makes the buffer to fill and freeze Tendermint
+    debug_tendermint = True  # This will cause more logging -> faster failure
+
+    @wait_for_node_to_run
+    def test_tendermint_buffer(self) -> None:
+        """Test Tendermint buffer"""
+
+        try:
+            # Give the test 30 seconds for it to throw a timeout
+            for _ in range(30):
+                # If it hangs for 5 seconds, we assume it's not working.
+                # Increasing the timeout should have no effect,
+                # the node is unresponsive at this point.
+                requests.get("http://localhost:26657/status", timeout=5)
+                time.sleep(1)
+        except requests.exceptions.Timeout:
+            raise AssertionError("Tendermint has timed out")
+
+    @classmethod
+    def teardown_class(cls) -> None:
+        """Teardown the test."""
+        cls.app_context.pop()
+        cls.tendermint_node.stop()
+        shutil.rmtree(cls.tm_home, ignore_errors=True, onerror=readonly_handler)
+
+
+class TestTendermintBufferWorking(BaseTendermintServerTest):
+    """Test Tendermint buffer"""
+
+    perform_monitoring = True
+    debug_tendermint = True
+
+    @wait_for_node_to_run
+    def test_tendermint_buffer(self) -> None:
+        """Test Tendermint buffer"""
+
+        # Give the test 60 seconds for it to work
+        for _ in range(60):
+            try:
+                res = requests.get("http://localhost:26657/status", timeout=5)
+                # We expect all responses to be OK
+                assert res.status_code == 200
+            except Exception as e:
+                raise AssertionError(e)
+
+            time.sleep(1)
```

Comment:
so I'm gonna comment it here but holds for the test above as well. We try 60 times, if not timeout for each attempt (might take upto 5 sec) then we're good. So can take upto 300 seconds. Is that what we want? then clarify in comments. 

However, I don't think the test is clear, I will explain in the review comment up next.

Reply:
To clarify what we have here, in my understanding when running the agent with `use_tendermint: True` we get the following log flow between tendermint and the agent: 
tendermint -> PIPE -> aea.

The [pipe has limited capacity](https://man7.org/linux/man-pages/man7/pipe.7.html#:~:text=A%20pipe%20has%20a%20limited,process%20does%20not%20remain%20blocked.), typically 16KB. When this limit is reached the process that writes to it *might* be blocked. 
It depends on the writing application (in our case tendermint), OS and the config of the `subprocess.popen()`call whether & when that might happen.
In our case we see this problem, at least on Debian distros, I haven't tried with windows. This blocking happens when the reader doesn't read and the pipe limit is reached. 

What we are trying to achieve with these tests is to show that when the pipe limit is reached, and the reader (AEA) isn't reading tendermint blocks.
So, what we do is we run the config as it is, with the small difference that we configure tendermint to use debug logging, because it will cause it to spit out more logs, which would fill the pipe faster, and ultimately cause the tendermint process to block faster. On my machine the failure happens in about ~6s. 

When reading is turned off (`perform_monitoring = False`), we give 30 seconds for tendermint to reach the pipe limit and block, we check whether tendermint has blocked if it doesn't return a 200 on the status check, we give 5s time for it to respond, which should be more than enough (it normally takes milliseconds).
The test shows this, tendermint takes more than 5s to respond, and we assume its blocked. Increasing the timeout should have no effect, although it would make the tests take longer, you can try out with a bigger timeout.

When reading is turned on (`perform_monitoring = True`), the pipe shouldn't get filled and tendermint shouldn't be blocked. To prove that this is the case, we give tendermint twice the time to reach the pipe limit and block. We can switch to 30s, that should have no difference, I chose for 60s just make sure.